olive/passes/onnx/nvmo_quantization.py

```diff
@@ -54,46 +55,386 @@ def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassCon
+        """Modify the model's opset to 21 if it's not already.
+
+        Args:
+            model_path (str): Path to the original ONNX model.
+            output_path (str): Path to save the converted model.
+
+        Returns:
+            str: Path to the saved model.
+
+        """
+        # Load the original ONNX model
+        model = onnx.load(model_path)
+
+        current_opset = {opset.domain: opset.version for opset in model.opset_import}
+
+        default_domain_version = current_opset.get("", 0)
+        if default_domain_version >= 21:
+            logger.info(
+                "Model already uses opset version %s for the default domain. Skip conversion.", default_domain_version
+            )
+            return model_path  # No conversion needed
+
+        logger.info("Converting model opset from %s to 21.", default_domain_version)
+
+        new_opset_imports = [
+            helper.make_opsetid("", 21),  # Default domain with opset version 21
+            helper.make_opsetid("com.microsoft", 1),  # Microsoft domain with version 1
+        ]
+
+        for domain, version in current_opset.items():
+            if domain not in ["", "com.microsoft"]:
+                new_opset_imports.append(helper.make_opsetid(domain, version))
+
+        # Create the updated model with new opset imports
+        updated_model = onnx.helper.make_model(model.graph, opset_imports=new_opset_imports)
+
+        # Define the external data file path (all tensors saved in one file)
+        external_data_path = os.path.basename(output_path) + "_data"
+
+        # Save the updated model with external data
+        onnx.save(
+            updated_model,
+            output_path,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+            location=external_data_path,
+        )
+
+        logger.info("Model opset successfully converted to 21 and saved to %s.", output_path)
+
+        # Return the path to the saved model
+        return output_path
+
     def _run_for_config(
         self, model: OliveModelHandler, config: Dict[str, Any], output_model_path: str
     ) -> OliveModelHandler:
+
+        temp_dir = tempfile.mkdtemp(prefix="modelopt_temp_")
         try:
-            from modelopt.onnx.quantization.int4 import quantize_int4  # type: ignore[import]
-        except ImportError as exc:
-            raise ImportError(
-                "Please install `olive-ai[nvmo]` or `nvidia-modelopt[onnx]` to use INT4 AWQ quantization!"
-            ) from exc
+            logger.info("Temporary directory created at %s.", temp_dir)
 
-        data_config = validate_config(config["data_config"], DataConfig)
-        calib_dataloader = data_config.to_data_container().create_dataloader()
+            temp_model_path = os.path.join(temp_dir, "model.onnx")
+            converted_model_path = self.convert_opset_to_21(model.model_path, temp_model_path)
```

Comment:
does the nvmo quantization script expect the model to have the opset 21 beforehand? if not, could we convert the opset to 21 on the output of `self.quantize_awq` so that we don't have to create a copy of the model?

Reply:
just check the quantize_int4 function. it accepts a modelproto so I don't think the save to disk is not necessary
? We could just pass the loaded model proto after the convert opset directly 
![image](https://github.com/user-attachments/assets/04dd80ff-9027-465c-8690-8c92e953f8bf)